src/script/descriptor.cpp

```diff
@@ -922,7 +925,8 @@ class DescriptorImpl : public Descriptor
                     if (!pubkey->ToNormalizedString(*arg, tmp, cache)) return false;
                     break;
                 case StringType::PRIVATE:
-                    if (!pubkey->ToPrivateString(*arg, tmp)) return false;
+                    assert(has_priv_key != nullptr);
```

Comment:
In 49f41141c674f83d0b66262a5b21aad623704264 "descriptor: ToPrivateString() pass if  at least 1 priv key exists"

If `has_priv_key` is never supposed to be null, then it should be a reference. Otherwise, this should be an `if`, we should not rely on caller behavior.